### ISSUE TITLE
unstable test in the StmtSummaryHistoryTable

### ISSUE THREAD

**Issue by hawkingrei** [pingcap/tidb#32004]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
     result.go:50: 
        	Error Trace:	result.go:50
        	            				cluster_tables_test.go:403
        	Error:      	Not equal: 
        	            	expected: "[select `insert` from `table`]\n[create table `table` ( `insert` int )]\n[set global `tidb_enable_stmt_summary` = ?]\n"
        	            	actual  : "[select `insert` from `table`]\n[create table `table` ( `insert` int )]\n[]\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	 [create table `table` ( `insert` int )]
        	            	-[set global `tidb_enable_stmt_summary` = ?]
        	            	+[]
        	            	 
        	Test:       	TestClusterTables/StmtSummaryHistoryTable
        	Messages:   	sql:select digest_text from information_schema.statements_summary_history;, args:[]
    --- FAIL: TestClusterTables/StmtSummaryHistoryTable (0.17s) 
```

**Comment by morgo** [comment]
We recently refactored how `tidb_enable_stmt_summary` works in https://github.com/pingcap/tidb/pull/30756 (with datarace fixed in https://github.com/pingcap/tidb/issues/31642 )

It is likely this only affects master

**Comment by hawkingrei** [comment]
> We recently refactored how `tidb_enable_stmt_summary` works in #30756 (with datarace fixed in #31642 )
> 
> It is likely this only affects master

Yes, I have changed the severity. so the robot will not add may-affects-xx labels.
